### DISCUSSION TITLE
WPF Webview2 application unittesting help wanted

### DISCUSSION THREAD

**Discussion by sensrsc** [MicrosoftEdge/WebView2Feedback#3959] [Q&A]
Hi Team,

We are currently developing our application based on WPF and using Webview2 for some automated web page navigation and interaction. We can successfully implement the feature by tutorials from Webview2 samples using `ExecuteScriptAsync` and corresponding lifecycle events, but now we are struggling on how to add unittests for implemented Webview Class.
Is there any tutorial or related article that helps on Webview2 application unittesting? or any suggestion on how can we get it started? 

Here are some basic environment information,
WPF based on .NETCore 5.0
Webview2

And we are currently using MSTEST as our unittesting library, any suggestion on unittesting libraries?

Suggestions and feedbacks are welcomed! Thank you!

**Comment by champnic** [comment]
Hey @sensrsc - If you are looking to automate the WebView2 for testing purposes, you can look into using WebDriver:
https://docs.microsoft.com/en-us/microsoft-edge/webview2/how-to/webdriver

If you are looking for true unit-testing, then I would assume you need to do some mocking of the WebView2 class.
